tools/project.py

```diff
@@ -344,6 +348,16 @@ def make_flags_str(flags: Optional[List[str]]) -> str:
     return " ".join(flags)
 
 
+def get_pch_out_name(config: ProjectConfig, pch: PrecompiledHeader) -> str:
+    pch_rel_path = Path(pch["source"])
+    pch_out_name = pch_rel_path.with_suffix(".mch")
+    # Use absolute path as a workaround to allow this target to be matched with absolute paths in depfiles.
+    #
+    # Without this any object which includes the PCH would depend on the .mch filesystem entry but not the
+    # corresponding Ninja task, so the MCH would not be implicitly rebuilt when the PCH is modified.
+    return os.path.abspath(config.out_path() / "include" / pch_out_name)
```

Comment:
Personally I would give pchs their own subdir just in case their name conflicts with anything else (since include dir already contains other things like assets, although none of them have .mch extension so it probably doesn't matter)
```suggestion
    return os.path.abspath(config.out_path() / "include" / "pch" / pch_out_name)
```

Reply:
I think we should leave it as-is (without the "pch" dir), otherwise we'd have to add the pch dir as a separate include